PXB-3450 PXB 8.4.0-2 Fails to Prepare Backup Taken with PXB 8.4.0-1

aybek opened the pull request:
Problem:
In PXB 8.4.0-2, a new parameter, 'opt_lock_ddl,' was introduced in the xtrabackup_info file.
When preparing a backup taken with PXB 8.4.0-1, PXB failed to locate the 'opt_lock_ddl' parameter in the xtrabackup_info file.

Solution:
Handle 'opt_lock_ddl' parameter as default 'ON' if not found in the xtrabackup_info file.

satya-bodapati reviewed (changes requested):
We cannot fix something that is already released :) 

Please handle the lack of opt_lock_ddl parameter and only if the version is 8.4.0-1

The fix should be released as part of next release 8.4.0-3

satya-bodapati commented on storage/innobase/xtrabackup/src/xtrabackup.cc:
```diff
@@ -2656,11 +2656,12 @@ static bool xtrabackup_read_info(char *filename) {
+    xb::warn() << "lock_ddl_type parameter not found, defaulting to 'ON'";
+    opt_lock_ddl = LOCK_DDL_ON;
+  } else {
+    /* used at log0recv.cc to apply or not file operations */
```
You mean, this option is used to handle the metadata files created by backup (lock-ddl=reduced)?

satya-bodapati commented on storage/innobase/xtrabackup/src/xtrabackup.cc:
```diff
@@ -3348,7 +3350,6 @@ void io_watching_thread() {
   io_watching_thread_running = false;
 }
 
-
```
can you plz fix the  line removals?

satya-bodapati commented on storage/innobase/xtrabackup/src/xtrabackup.cc:
```diff
@@ -3575,7 +3576,6 @@ Initializes the I/O and tablespace cache subsystems. */
 static bool xb_fil_io_init(void)
 /*================*/
 {
-
```
new line removed here

satya-bodapati commented on storage/innobase/xtrabackup/src/xtrabackup.cc:
```diff
@@ -3714,7 +3714,6 @@ void xb_data_files_close(void)
 
   fil_close();
 
-
```
newline removed here, restore

satya-bodapati commented on storage/innobase/xtrabackup/src/xtrabackup.cc:
```diff
@@ -6898,7 +6897,6 @@ static void xtrabackup_prepare_func(int argc, char **argv) {
 
   srv_apply_log_only = (bool)xtrabackup_apply_log_only;
 
-
```
newline removed here, restore

satya-bodapati reviewed (changes requested):
Please fix them and squash.

Also please create a trunk version of the fix.

aybek committed:
PXB-3450 PXB 8.4.0-2 Fails to Prepare Backup Taken with PXB 8.4.0-1
Problem:
In PXB 8.4.0-2, a new parameter, 'opt_lock_ddl,' was introduced in the xtrabackup_info file.
When preparing a backup taken with PXB 8.4.0-1, PXB failed to locate the 'opt_lock_ddl' parameter in the xtrabackup_info file.

Solution:
Handle 'opt_lock_ddl' parameter as default 'ON' if not found in the xtrabackup_info file.

satya-bodapati reviewed (approved):
LGTM!
